.github/workflows/build-test.yml

```diff
@@ -115,7 +122,10 @@ jobs:
     needs: pre-flight-check
     steps:
       - uses: actions/checkout@v3
-      - run: ./tools/ci/build-test-macos-with-sanitizers.sh
+      - name: Run Address Sanitizer
+        run: ./tools/ci/build-test-macos-with-sanitizers.sh asan
+      - name: Run Thread Sanitizer
+        run: ./tools/ci/build-test-macos-with-sanitizers.sh tsan
```

Comment:
@ibrhmkuru The current OS for Mac is using [macOS 11.7](https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md) with Clang 13 as default toolchain.

On MacOS 12 we have Clang 14 installed: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md

Would you test that on MacOS 12? You just need to change this statement above: `runs-on: macos-latest` to `runs-on: macos-12`. And let's see about the differences.